buildSrc/src/main/java/com/google/firebase/gradle/plugins/ProjectUtils.kt

```diff
@@ -23,3 +29,45 @@ fun toBoolean(value: Any?): Boolean {
     val trimmed = value?.toString()?.trim()?.toLowerCase()
     return "true" == trimmed || "y" == trimmed || "1" == trimmed
 }
+
+/**
+ * Shorthand for [BaseExtension.getBootClasspath]
+ *
+ * This method assumes your project has the [BaseExtension] extension,
+ * as it uses [ExtensionContainer.getByType]
+ */
+val Project.bootClasspath: List<File>
+    get() = extensions.getByType<BaseExtension>().bootClasspath
+
+/**
+ * Finds or creates the javadocClasspath [Configuration].
+ */
+val Project.javadocConfig: Configuration
+    get() = configurations.findByName("javadocClasspath") ?: configurations.create("javadocClasspath")
+
+/**
+ * Finds or creates the dackkaArtifacts [Configuration].
+ *
+ * Used to fetch the dackka-fat jar at runtime versus needing to explicitly specify it in project
+ * dependencies.
+ */
+val Project.dackkaConfig: Configuration
+    get() =
+        configurations.findByName("dackkaArtifacts") ?: configurations.create("dackkaArtifacts") {
+            dependencies.add(this@dackkaConfig.dependencies.create("com.google.devsite:dackka-fat:1.0.1"))
+        }
+
+/**
+ * Fetches the jars of dependencies associated with this configuration through an artifact view.
+ *
+ * Excluding androidx.annotation:annotation artifacts.
+ */
+fun Configuration.getJars() = incoming.artifactView {
+    attributes {
+        // replace value with android-class instead of jar after agp upgrade
+        attribute(Attribute.of("artifactType", String::class.java), "jar")
+    }
+    componentFilter {
+        !it.displayName.startsWith("androidx.annotation:annotation:")
+    }
```

Comment:
Do you know if this is actually needed? I can't recall why we filter it out.

Reply:
Actually, I just tested your chance locally and it appears that this exclude causes nullability issues, i.e. with it all `@Nullable/@NotNull` annotations on java code are ignored and it cause all kotlin doc to have "platform types", i.e. `fun foo(bar: Bar!): Baz!` instead of `fun foo(bar: Bar?): Baz`. I think we want to remove this line(removing this locally fixed the nullability issue for me).

Reply:
No, I have no idea. Although- it seems to generate just fine. I'll remove it for now, and we can revisit this if we end up needing it removed again.

Reply:
I think I remember now, in the old dokka having nullability annotations on the classpath would result in annotations literally propagating to the kotlindoc, i.e. it would render with `@NonNull`, so I removed it and it did not cause any issues with kotlin nullability, it looks like it's no longer the case, so yeah let's remove this